config/toml.go

```diff
@@ -17,34 +27,158 @@ func EnsureRoot(rootDir string) {
 
 	configFilePath := path.Join(rootDir, "config.toml")
 
-	// Write default config file if missing.
-	if !cmn.FileExists(configFilePath) {
-		// Ask user for moniker
-		// moniker := cfg.Prompt("Type hostname: ", "anonymous")
-		cmn.MustWriteFile(configFilePath, []byte(defaultConfig("anonymous")), 0644)
+	if cmn.FileExists(configFilePath) {
+		// TODO: Prompt user to replace if config file already exists.
+	} else {
+		// Ask user for moniker [moniker := cfg.Prompt("Type hostname: ", "anonymous")] (TODO)
+		// Then write the default config from template
+		writeConfigFile(configFilePath)
+	}
+}
+
+// XXX: this func should probably be called by cmd/tendermint/commands/init.go
+// alongside the writing of the genesis.json and priv_validator.json
```

Comment:
thoughts? seems awkward to not have the `config.toml` be initialized alongside `genesis.json` and `priv_validator.json` [here](https://github.com/tendermint/tendermint/blob/master/cmd/tendermint/commands/init.go#L23-L36)